odxtools/parameters/lengthkeyparameter.py

```diff
@@ -45,23 +45,14 @@ def is_optional(self):
         return True
 
     def get_coded_value_as_bytes(self, encode_state: EncodeState) -> bytes:
-        physical_value = encode_state.length_keys[self.short_name]
+        physical_value = encode_state.parameter_values.get(self.short_name, 0)
 
         bit_pos = self.bit_position or 0
         dop = super().dop
         assert dop is not None, f"A DOP is required for length key parameter {self.short_name}"
         return dop.convert_physical_to_bytes(physical_value, encode_state, bit_position=bit_pos)
 
     def encode_into_pdu(self, encode_state: EncodeState) -> bytes:
-        physical_value = encode_state.parameter_values.get(self.short_name)
-
-        if physical_value is None:
-            raise TypeError(f"A value for the length key '{self.short_name}'"
-                            f" must be specified.")
-
-        # Set the value of the length key in the length key dict.
-        encode_state.length_keys[self.short_name] = physical_value
-
         return super().encode_into_pdu(encode_state)
```

Comment:
This makes the overload of this function useless, it could be just removed

Reply:
yes. I thought it would improve the visibility (`get_coded_value_as_bytes()` is not supposed to be called externally if I understand this coed correctly). The en-/decoding code should be fundamentally refactored soon because it is currently quite a bowl of spaghetti IMO...